### PR TITLE
Add keep-current-tfms option

### PR DESCRIPTION
**Pull request by cartermp**
This effectively lets you say, "I am just converting to the .NET SDK for now". Some notes:

* This will still add the desktop SDK and UseWinForms/UseWPF if you have the references
* I tested this out manually and there doesn't appear to be anything inherently wrong with referencing the desktop SDK while targeting .NET Framework, but you may not get the same APIs
* I tested this out on a large internal project and it seems okay

This required some small refactoring around determining the project TFM. This also moves converting to PackageReference to be first, since that's what you'd normally due if you were migrating manually. If we ever wanted to introduce a "just convert my packages" option then this would enable that

### CODE REVIEW

File: src/MSBuild.Abstractions/MSBuildHelpers.cs [hunk 1]
```diff
@@ -142,12 +142,10 @@ public static bool ConditionToDimensionValues(string condition, out ImmutableDic
...
-            return tfm.ContainsIgnoreCase(MSBuildFacts.NetstandardPrelude)
-                || tfm.ContainsIgnoreCase(MSBuildFacts.NetcoreappPrelude)
-                ? false
-                : !tfm.StartsWith("net", StringComparison.OrdinalIgnoreCase)
-                    ? false
-                    : tfm.StartsWith(MSBuildFacts.LowestFrameworkVersionWithSystemValueTuple);
+            return !tfm.ContainsIgnoreCase(MSBuildFacts.NetstandardPrelude)
+                && !tfm.ContainsIgnoreCase(MSBuildFacts.NetcoreappPrelude)
+                && (tfm.StartsWith("net", StringComparison.OrdinalIgnoreCase)
+                    && tfm.StartsWith(MSBuildFacts.LowestFrameworkVersionWithSystemValueTuple));
```
Comment (cartermp) [inline src/MSBuild.Abstractions/MSBuildHelpers.cs:148]: Roslyn suggested that I do this and so I did

File: src/try-convert/Program.cs [hunk 2]
```diff
@@ -32,16 +32,17 @@ private static async Task<int> Main(string[] args)
                 .AddOption(new Option(new[] { "-p", "--project" }, "The path to a project to convert") { Argument = new Argument<string?>(() => null) })
                 .AddOption(new Option(new[] { "-w", "--workspace" }, "The solution or project file to operate on. If a project is not specified, the command will search the current directory for one.") { Argument = new Argument<string?>(() => null) })
                 .AddOption(new Option(new[] { "-m", "--msbuild-path" }, "The path to an MSBuild.exe, if you prefer to use that") { Argument = new Argument<string?>(() => null) })
-                .AddOption(new Option(new[] { "-tfm", "--target-framework" }, "The name of the framework you would like to upgrade to") { Argument = new Argument<string?>(() => null) })
+                .AddOption(new Option(new[] { "-tfm", "--target-framework" }, "The name of the framework you would like to upgrade to. Useful if you'd prefer not to change TFMs to .NET Core just yet.") { Argument = new Argument<string?>(() => null) })
```
Comment (cartermp) [inline src/try-convert/Program.cs:35]: This kind of obviates the `-tfm` option, I'm just putting this up here now but I wanna think about this one before doing anything else

Comment (cartermp) [reply src/try-convert/Program.cs]: Thought about it more, it's not obviated

File: src/MSBuild.Abstractions/MSBuildHelpers.cs (hunk 1 above)
Comment (jmarolf) [reply src/MSBuild.Abstractions/MSBuildHelpers.cs:148]: Hope I never need to debug this :(

---
In reply to: [419817573](https://github.com/dotnet/try-convert/pull/285#discussion_r419817573) [](ancestors = 419817573)

**Pull request merged**